src/legacy/core_plugins/dashboard_embeddable_container/public/np_ready/public/lib/actions/change_view_action.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { CoreStart } from 'src/core/public';
+
+import { IEmbeddable } from '../../../../../../embeddable_api/public/np_ready/public';
+import { DASHBOARD_CONTAINER_TYPE, DashboardContainer } from '../embeddable';
+import {
+  IAction,
+  IncompatibleActionError,
+} from '../../../../../../../../../src/plugins/ui_actions/public';
+import { NotificationsStart } from '../../../../../../../../core/public';
+import { openChangeViewFlyout } from './open_change_view_flyout';
+
+export const CHANGE_VIEW_ACTION = 'changeView';
+
+function isDashboard(embeddable: IEmbeddable): embeddable is DashboardContainer {
+  return embeddable.type === DASHBOARD_CONTAINER_TYPE;
+}
+
+interface ActionContext {
+  embeddable: IEmbeddable;
+}
+
+export class ChangeViewAction implements IAction<ActionContext> {
+  public readonly type = CHANGE_VIEW_ACTION;
+  public readonly id = CHANGE_VIEW_ACTION;
+  public order = 11;
+
+  constructor(
+    private core: CoreStart,
+    private savedobjectfinder: React.ComponentType<any>,
+    private notifications: NotificationsStart
+  ) {}
+
+  public getDisplayName({ embeddable }: ActionContext) {
+    if (!embeddable.parent || !isDashboard(embeddable.parent)) {
+      throw new IncompatibleActionError();
+    }
+    return i18n.translate('dashboardEmbeddableContainer.panel.removePanel.replaceView', {
+      defaultMessage: 'Replace visualization',
+    });
+  }
+
+  public getIconType({ embeddable }: ActionContext) {
+    if (!embeddable.parent || !isDashboard(embeddable.parent)) {
+      throw new IncompatibleActionError();
+    }
+    return 'kqlOperand';
+  }
+
+  public async isCompatible({ embeddable }: ActionContext) {
+    return Boolean(embeddable.parent && isDashboard(embeddable.parent));
```

Comment:
should check for edit mode as well, we don't want it showing up in view mode.

Reply:
Ok, right.